linux/src/async_socket_linux.c

```diff
@@ -409,9 +421,17 @@ static int initialize_global_thread(void)
                 interlocked_decrement(&g_thread_access_cnt);
             }
         }
+        (void)interlocked_decrement(&g_epoll_access);
+        wake_by_address_single(&g_epoll_access);
     }
     else
     {
+        int32_t value;
+        while ((value = interlocked_add(&g_epoll_access, 0)) != 0)
+        {
+            (void)wait_on_address(&g_epoll_access, value, UINT32_MAX);
+        }
```

Comment:
In the light of "let's move the epoll+thread to another unit I think this counting would not be needed at all (would be handled by this other unit fully.

Reply:
So maybe we should remove this counting altogether.

Reply:
Yes, I will move this, but I'd like to move it in the next PR.